update scheduling guidances (#192)
quicwg/multipath: draft-ietf-quic-multipath.md

The transmission of QUIC packets on a regular QUIC connection is regulated
by the arrival of data from the application and the congestion control
scheme. QUIC packets can only be sent when the congestion window is open.

> 

Reviewer: Can we say this a little better? ACK frames, for example, are not subject to the congestion control. The proper statement would be:
~~~suggestion
QUIC packets that increase the number of bytes in flight can only be sent when the congestion window is open.
~~~
But that's a bit tautological. One could just drop the whole sentence, because that's exactly what the word "regulated" above means.

Author: You're right. Your suggestion above is good (committing now). But yeah, this sentence may not be necessary in the end...

> scheme. QUIC packets can only be sent when the congestion window is open.
> QUIC packets that increase the number of bytes in flight can only be sent when the congestion window is open.

Reviewer 2: ```suggestion
scheme. QUIC packets that increase the number of bytes in flight can only be sent
when the congestion window allows it.
```

Multipath QUIC implementations also need to include a packet scheduler
that decides, among the paths whose congestion window is open, the path
over which the next QUIC packet will be sent. Most frames, including
control frames (PATH_CHALLENGE and PATH_RESPONSE being the notable
exceptions), can be sent and received on any active path. The scheduling

> is a local choice, based on the preferences of the application and the

Reviewer 3: ```suggestion
is a local decision, based on the preferences of the application and the
```

implementation.

Note that this implies that an endpoint may send and receive ACK_MP
frames on a path different from the one that carried the acknowledged
packets. A reasonable default consists in sending ACK_MP frames on the
path they acknowledge packets, but the receiver must not assume its

> peer will do so.

Author: For the normative text, I was actually proposing to remove the text about endpoints that SHOULD send ACK_MP on the same path. But this would not be part of this PR.

Regarding your first point, I'm not sure sending no ACK at all on the same path necessarily impact RTT estimation in a bad way. As long as you keep consistency in sending your ACKs, this should not be a big issue. Also note that with network paths having asymmetrical characteristics in forward or backward directions, it might be desirable from a performance viewpoint to send ACKs on some specific paths. See also #190. My main point is that we should not rely on specific scheduling from the peer to make Multipath QUIC work.

Regarding your second point, I'm not against keeping the reference. However, this other draft is not adopted (and currently expired), not sure how this would be a problem in the publication process.

> peer will do so.

Reviewer: I agree with the proposal. It does remove text, but less text is better. And it says exactly what we want, i.e., scheduling of packets is up to the implementation.

In particular, I like the removal of the reference to time stamps. The draft is expiring, and despite efforts nobody is picking it up. Rather not have a reference to an expired draft in a proposed standard.

> peer will do so.

Reviewer 2: Okay to not say much about RTT estimates now. We can revisit that later if anything needed. Might be good to review the text we have about RTT estimation first.

About the reference, having a reference to an individual draft or even expired draft is no problem as long as it is informational. However, maybe we have a nice research paper we can refer instead? I can try to find something but @qdeconinck I'm sure you also have a good overview.

> peer will do so.

Author: To the best of my knowledge, I saw a lot of research works about Multipath TCP schedulers, mostly to achieve bandwidth aggregation in different network situations. There are only a few papers out discussing Multipath QUIC schedulers, but they are most of the time very focused on a specific case and often based on an old Multipath Google QUIC (2017) prototype -- XQUIC being one of the notable exceptions.

If there is no issue in keeping the informational draft about scheduling and if you still want to have a reference discussing different possible strategies, then let's do that.

> peer will do so.

Reviewer 2: I know it is merged now, which is fine. However, having a reference to different scheduling strategies I think is still useful. Even if those papers are about TCP, the scheduling is usually not that much protocol dependent (except that in addition can/should consider streams in QUIC). But maybe the individual draft is still the best reference as it describes different approach in an protocol independent way.
